embed/etcd.go

```diff
@@ -335,8 +338,17 @@ func (e *Etcd) serve() (err error) {
 		// read timeout does not work with http close notify
 		// TODO: https://github.com/golang/go/issues/9524
 		go func(s *serveCtx) {
-			e.errc <- s.serve(e.Server, ctlscfg, v2h, e.errc)
+			e.errHandler(s.serve(e.Server, ctlscfg, v2h, e.errHandler))
 		}(sctx)
 	}
 	return nil
 }
+
+func (e *Etcd) errHandler(err error) {
+	select {
```

Comment:
if both `stopc` and `errc` are not blocking, the select will randomly choose a case to process (if i understand channel correctly).  If the chosen case is ` e.errc <- err:`, that would be a problem.